lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -192,23 +197,30 @@ mkTx
     -> CoinSelection
     -> Either ErrMkTx (Tx, SealedTx)
 mkTx proxy pm (TxPayload certs mkExtraWits) timeToLive (rewardAcnt, pwdAcnt) keyFrom cs = do
+    {-
     let wdrls = mkWithdrawals
             proxy
             (toChimericAccountRaw . getRawKey . publicKey $ rewardAcnt)
             (withdrawal cs)
 
     let unsigned = mkUnsignedTx timeToLive cs wdrls certs
 
-    wits <- case (txWitnessTagFor @k) of
+    sealed <- case (txWitnessTagFor @k) of
         TxWitnessShelleyUTxO -> do
+            let paymentKey = _ (getRawKey k, pwd)
+            let stakeKey
+                     | null wdrls = []
+                     | otherwise = [_ (getRawKey rewardAcnt, pwdAcnt)]
+
+            tx <- signShelleyTransaction unsigned (paymentKey:stakeKey)
```

Comment:
This seems weird to me, what was the problem here?

Reply:
Also, it seems you almost went in the direction of creating a byron vs Shelley tx based on the key/witness tag, but we may spend from byron addresses in shelley.

 So we need to either on the era we're in (or if there's something in cardano-api that does that automatically)?

Reply:
Basically what happened is there were functions removed that we were using, so I attempted to use the new API and broke it.